### PR TITLE
Add scheduled-job and adhoc-job charts

### PR DESCRIPTION
**Pull request by chrisghill**
closes: https://github.com/massdriver-cloud/massdriver/issues/901

### CODE REVIEW

**Comment by coryodaniel** [comment]
Looks good, two notes though:

* We should bake these into the cli using embed.FS (see bundle gen for an example) under `templates/helm/NAME` so we have a nice layout to support `templates/lambda/NAME` in the future
* When you make the selector in the CLI prompt, let's not use the word "Deployment" since we are going to need to reserve that for CD, maybe 'template' or Application Type?

**Comment by coryodaniel** [comment]
One note to save you from go-misery. 

`{{` and `}}` do _not_ make the go template engine happy since they are what it uses to interpolate go variables. 

I'd avoid running anything helm charts through there if you don't need to (I'm assuming all the interpolation would happen in values.yaml) otherwise you'll need to go escape every `{{` with `{{"{{"}}`.

Given that, we should probably standardize on `boilerplate` and `templates` to save from escape hell.

* `generator/boilerplates/helm/NAME/a-file-that-doesnt-need-to-be-interpolated`
* `generator/templates/helm/NAME/a-file-that-does-need-to-be-interpolated`

That will save the generator from having a bunch of code that knows which files to run through the template engine or not and instead make it a convention of the _application templates_

**Comment by chrisghill** [comment]
> * We should bake these into the cli using embed.FS (see bundle gen for an example) under `templates/helm/NAME` so we have a nice layout to support `templates/lambda/NAME` in the future
I was actually planning to do a straight up `helm pull` via the helm SDK to pull down the chart instead of baking it into the CLI. What do you think?

> * When you make the selector in the CLI prompt, let's not use the word "Deployment" since we are going to need to reserve that for CD, maybe 'template' or Application Type?
Agreed! 👍

**Comment by coryodaniel** [comment]
I guess it depends on how far out we think lambda is?

What would be sick for customizability but a PITA
 
If we renamed the repo to “templates” and put all the apps under “apps”

when you do ‘mass app gen’ it defaults to our repo, and presents a prompt off all the dirs under “apps” you select it and it renders that into place

then people could do custom repos down the line w ‘mass app gen -r m path to my repo’

**Comment by chrisghill** [comment]
Yeah, I was thinking something very similar 👍  I'll get this merged and working for these 3, then we can circle back to this and figure it our for serverless functions/containers when we get there (probably soon).

File: charts/adhoc-job/templates/job.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+apiVersion: batch/v1beta1
+kind: Job
+metadata:
+  name: {{ include "adhoc-job.timestamped-name" . }}
```
Comment (coryodaniel) [inline charts/adhoc-job/templates/job.yaml:4]: Should we use generate name instead for ad hoc jobs?

**Pull request merged**